tools/src/test/java/org/apache/kafka/tools/ProducerPerformanceTest.java

```diff
@@ -0,0 +1,152 @@
+        file.deleteOnExit();
+        Files.write(file.toPath(), contents.getBytes());
+        return file;
+    }
+
+    @Test
+    public void testReadPayloadFile() throws Exception {
+        File payloadFile = createTempFile("Hello\nKafka");
+        String payloadFilePath = payloadFile.getAbsolutePath();
+        String payloadDelimiter = "\n";
+
+        List<byte[]> payloadByteList = ProducerPerformance.readPayloadFile(payloadFilePath, payloadDelimiter);
+
+        assertEquals(2, payloadByteList.size());
+        assertEquals("Hello", new String(payloadByteList.get(0)));
+        assertEquals("Kafka", new String(payloadByteList.get(1)));
+    }
+
+    @Test
+    public void testReadProps() throws Exception {
+        
+        List<String> producerProps = Collections.singletonList("bootstrap.servers=localhost:9000");
+        String producerConfig = createTempFile("acks=1").getAbsolutePath();
+        String transactionalId = "1234";
+        boolean transactionsEnabled = true;
+
+        Properties prop = ProducerPerformance.readProps(producerProps, producerConfig, transactionalId, transactionsEnabled);
+
+        assertNotNull(prop);
+        assertEquals(5, prop.size());
+    }
+
+    @Test
+    public void testNumberOfCallsForSendAndClose() throws IOException {
+
+        doReturn(null).when(producerMock).send(any(), any());
+        doReturn(producerMock).when(producerPerformanceSpy).createKafkaProducer(any(Properties.class));
+
+        String[] args = new String[] {"--topic", "Hello-Kafka", "--num-records", "5", "--throughput", "100", "--record-size", "100", "--producer-props", "bootstrap.servers=localhost:9000"};
+        producerPerformanceSpy.start(args);
+        verify(producerMock, times(5)).send(any(), any());
+        verify(producerMock, times(1)).close();
+    }
+
+    @Test
+    public void testUnexpectedArg() {
+
+        String[] args = new String[] {"--test", "test", "--topic", "Hello-Kafka", "--num-records", "5", "--throughput", "100", "--record-size", "100", "--producer-props", "bootstrap.servers=localhost:9000"};
+        ArgumentParser parser = ProducerPerformance.argParser();
+        ArgumentParserException thrown = assertThrows(ArgumentParserException.class, () -> parser.parseArgs(args));
+        assertEquals("unrecognized arguments: '--test'", thrown.getMessage());
+    }
+
+    @Test
+    public void testGenerateRandomPayloadByPayloadFile() {
+
+        Integer recordSize = null;
+        String inputString = "Hello Kafka";
+        byte[] byteArrray = inputString.getBytes(StandardCharsets.UTF_8);
+        List<byte[]> payloadByteList = new ArrayList<>();
+        payloadByteList.add(byteArrray);
+        byte[] payload = null;
+        Random random = new Random(0);
+
+        payload = ProducerPerformance.generateRandomPayload(recordSize, payloadByteList, payload, random);
+        assertEquals("Hello Kafka", new String(payload));
```

Comment:
Could you replace constant string by `inputString`?

Reply:
Done.